src/OrbitQt/ProfilingTargetDialog.cpp

```diff
@@ -495,11 +492,16 @@ void ProfilingTargetDialog::ConnectToLocal() {
 void ProfilingTargetDialog::SetupLocalProcessManager() { SetupProcessManager(local_grpc_channel_); }
 
 void ProfilingTargetDialog::SetTargetAndInitialState(StadiaTarget target) {
-  process_manager_ = std::move(target.process_manager_);
-  process_ = std::move(target.process_);
+  if (target.connection_.GetGrpcChannel()->GetState(false) == GRPC_CHANNEL_READY) {
+    ui_->stadiaWidget->SetConnection(std::move(target.connection_));
+
+    process_manager_ = std::move(target.process_manager_);
+    process_ = std::move(target.process_);
+
+    state_stadia_.setInitialState(&state_stadia_connected_);
+    state_stadia_history_.setDefaultState(&state_stadia_connected_);
+  }
 
-  state_stadia_.setInitialState(&state_stadia_connected_);
-  state_stadia_history_.setDefaultState(&state_stadia_connected_);
   state_machine_.setInitialState(&state_stadia_);
```

Comment:
May be we can also rename the function `SetInitalStateFromTarget` as `SetStateMachineInitalStateFromTarget` and move this line for setting initial state for `state_machine_` there. If so, we can also have an early return in this method.

Reply:
I changed it. ptal

Reply:
Looks good. Thanks for making the change : )